src/CommercetoolsAuth0.ts

```diff
@@ -222,6 +228,38 @@ export class CommercetoolsAuth0 {
     return null
   }
 
+  /**
+   * Return the customer details from the Commercetools graphQl response
+   *
+   * @param id - Commercetools customer id
+   */
+  public async getCustomerDetails(customerId: string): Promise<Omit<CustomerDetails, 'isNewCustomer'>> {
+    let response
+    try {
+      response = await this.client.graphql({
+        data: {
+          query: `{
+            customer(id: "${customerId}") {
+              firstName
+              lastName
+              customerGroup {
+                key
+              }
+            }
+          }`,
+        },
+      })
+    } catch (error) {
+      console.log(`Error retrieving customer details for customer Id [${customerId}]`)
+    }
+    return {
+      id: customerId,
+      firstName: response?.data?.customer?.firstName || '',
+      lastName: response?.data?.customer?.lastName || '',
+      groupKey: response?.data?.customer?.customerGroup?.key || '',
```

Comment:
I don't think the `|| ''` are required here. The `CustomerDetails` type defines these such that `undefined` is valid. Optionally, you could change the `CustomerDetails` type to not allow undefined properties. If you do that, I would change `groupKey` to `|| null` so that `null` clearly indicates that there's no value.